lib/vagrant-libvirt/templates/domain.xml.erb

```diff
@@ -81,8 +81,18 @@
     <cmdline><%= @cmd_line %></cmdline>
 <%- if @dtb -%>
     <dtb><%= @dtb %></dtb>
+<% end -%>
+<%- if @dmi_system_serial -%>
+    <smbios mode='sysinfo'/>
 <% end -%>
   </os>
+<%- if @dmi_system_serial -%>
+  <sysinfo type='smbios'>
+    <system>
+      <entry name='serial'><%= @dmi_system_serial %></entry>
```

Comment:
Nearly forgot, one possible improvement would be to maintain an internal list of recognized entries that can automatically be assigned a corresponding section if there is concern about making the end user interface as simple as possible.

e.g. support the following:
```
require 'securerandom'
Vagrant.configure("2") do |config|
  config.vm.provider :libvirt do |libvirt|
    libvirt.sysinfo :serial => SecureRandom.alphanumeric(8).upcase
  end
end
```

Where the function:
```
def sysinfo(options = {})
  entry_sections = {
    :serial => :system,
  }

  section = options.del(:section)
  entry = Hash[*options.first]
  raise "Section required for sysinfo entry #{entry.keys.first}" if section.nil? or entry_sections.fetch(entry.keys.first).nil?

  sysinfo_options[section] ||= {}
  sysinfo_options[section].merge!(entry)
end
```

Possibly it might make it more robust to support explicit entry name and value with section being optional to avoid collisions and store as a hash:
```
Vagrant.configure("2") do |config|
  config.vm.provider :libvirt do |libvirt|
    libvirt.sysinfo :name => 'serial', :value => SecureRandom.alphanumeric(8).upcase
  end
end
```

Either way I think need to avoid needing one config option per sysinfo entry as more are added.